### PR TITLE
[FIX] l10n_gcc_invoice: payment terms not shown in english/arabic

### PR DESCRIPTION
**Pull request by agr-odoo**
Have a SA company
In Accounting settings activate invoice terms
Add a payment terms string in english, add the arabic translation
Create an invoice
Click preview

Issue: Payment terms will be shown in english twice

This occurs because, even if `narration` field is translatable the translation
 is not automatically copied when the field is copied from the company `invoice_terms`.

opw-3530811

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by agr-odoo** [comment]
@nle-odoo 
This seems to work but it does not seems very clean. What do you think about it?

**Review by nle-odoo — APPROVED** [review]
yes, seems rather ugly, maybe we do other hackiness to have a little less code and excision have something like in the original method:

```diff
+    def _copy_company_terms(self, lang):
+        self.narration = self.company_id.with_context(lang=lang).invoice_terms if not is_html_empty(self.company_id.invoice_terms) else ''
+
     @api.depends('move_type', 'partner_id', 'company_id')
     def _compute_narration(self):
         use_invoice_terms = self.env['ir.config_parameter'].sudo().get_param('account.use_invoice_terms')
@@ -1347,8 +1359,9 @@ class AccountMove(models.Model):
                 move.narration = False
             else:
                 lang = move.partner_id.lang or self.env.user.lang
-                if not move.company_id.terms_type == 'html':
-                    narration = move.company_id.with_context(lang=lang).invoice_terms if not is_html_empty(move.company_id.invoice_terms) else ''
+                if move.company_id.terms_type == 'html':
+                    move._copy_company_terms(lang)
+                    continue
                 else
```
then in the override something like

```diff
+    # override in l10n_gcc_invoice
+    def _copy_company_terms(self, lang):
+        if [possible other condition]:
+            for lang in dict(self.env['res.lang'].get_installed()):
+                super(AccounMove, self.with_context(lang))._copy_company_terms(lang)
+        else:
+            super()._copy_company_terms(lang)
```
or something like you did but I don't know these methods too much =/

File: addons/l10n_gcc_invoice/models/account_move.py
Comment (agr-odoo) [reply addons/l10n_gcc_invoice/models/account_move.py:39]: compute method is running on new record, when we don't have the id and we can't update the translation, so the real record will not have the translation once created without the override

**Review by clbr-odoo — APPROVED** [review]
Hello everyone, @agr-odoo, @nle-odoo @HydrionBurst 

I'll move on with this PR unless you have something more to say ? 
We all agree it's hacky but working, better than doing nothing I would say :slightly_smiling_face:

**Comment by csnauwaert** [comment]
Ok let's move on with this one :)
@HydrionBurst I still see a few comments from you, is this current version ok with you? Can we FW port it to master or has anything been fixed/changed in the meantime?
@odoo/rd-security can we get your blessing for a Markup usage in a test?

Thanks all :)

**Review by rco-odoo — COMMENTED** [review]
I don't understand how this could _ever_ have worked before (even before Odoo 16).  This use-case (computed translated field) has never been supported by the ORM.

**Comment by clbr-odoo** [comment]
Hello @agr-odoo, note that Accounting agrees with this PR. 
I let you see with Hydrion and RCO if that's good to go for them, anyone can r+ if so. :slightly_smiling_face:

**Comment by nle-odoo** [comment]
robodoo delegate+

**Comment by agr-odoo** [comment]
@robodoo r+

**Pull request closed without merging**
